src/CSnakes.Runtime/Python/PyObject.cs

```diff
@@ -658,8 +659,10 @@ internal object As(Type type)
                 var t when t.IsAssignableTo(typeof(ITuple)) => PyObjectTypeConverter.ConvertToTuple(this, t),
                 var t when t.IsAssignableTo(typeof(IGeneratorIterator)) => PyObjectTypeConverter.ConvertToGeneratorIterator(this, t),
                 var t when t.IsAssignableTo(typeof(ICoroutine)) => PyObjectTypeConverter.ConvertToCoroutine(this, t),
+                var _ when Nullable.GetUnderlyingType(type) != null => IsNone() ? null : As(Nullable.GetUnderlyingType(type)!),
```

Comment:
Remove duplicate call to `Nullable.GetUnderlyingType`:

```suggestion
                var t when Nullable.GetUnderlyingType(t) is { } vt => IsNone() ? null : As(vt)
```